src/pipelinedefinition.cpp

```diff
@@ -972,10 +983,22 @@ Status PipelineDefinition::validateNodes(ModelManager& manager) {
     return StatusCode::OK;
 }
 
-Status PipelineDefinition::getInputsInfo(tensor_map_t& inputsInfo, const ModelManager& manager) const {
+const tensor_map_t PipelineDefinition::getInputsInfo() const {
+    std::shared_lock lock(metadataMtx);
+    tensor_map_t copy = inputsInfo;
```

Comment:
Here you copy the container but it consists of shared pointers. This is shallow copy.